src/Parser/RestoDruid/Modules/Features/StatWeights.js

```diff
@@ -0,0 +1,256 @@
+
+  totalAdjustedHealing = 0; // total healing after excluding 'multiplier' spells like Leech / Velens
+
+  // These are the total healing that would be gained if their respective stats ratings were increased by one.
+  totalOneInt = 0;
+  totalOneCrit = 0;
+  totalOneHasteHpm = 0;
+//  totalOneHasteHpct = 0;
+  totalOneMastery = 0;
+  totalOneVers = 0;
+  totalOneLeech = 0;
+
+  on_initialized() {
+    this.concordanceAmount = 4000 + (this.combatants.selected.traitsBySpellId[SPELLS.CONCORDANCE_OF_THE_LEGIONFALL_TRAIT.id] || 0) * 300; // TODO remove when stat tracker added
+  }
+
+  on_byPlayer_heal(event) {
+    const healVal = new HealingValue(event.amount, event.absorbed, event.overheal);
+    this._handleHeal(event, healVal);
+  }
+
+  on_byPlayer_absorbed(event) {
+    const healVal = new HealingValue(event.amount, 0, 0);
+    this._handleHeal(event, healVal);
+  }
+
+  _handleHeal(event, healVal) {
+    const target = this.combatants.getEntity(event);
+    if(target === null) {
+      return;
+    }
+
+    const amount = healVal.effective;
+    const spellInfo = getSpellInfo(event.ability.guid);
+
+    // LEECH //
+    // We have to calculate leech weight differently depending on if we already have any leech rating.
+    // Leech is marked as a 'multplier' heal, so we have to check it before we do the early return below
+    const hasLeech = this.combatants.selected.leechRating > 0; // TODO replace when dynamic stats
+    let oneLeech = 0;
+    if(hasLeech && spellInfo.isLeech && !healVal.overheal) {
+      oneLeech = amount / this.combatants.selected.leechRating; // TODO replace when dynamic stats
+    } else if(!hasLeech) {
+      // TODO this will be a pain to implement
+    }
+    this.totalOneLeech += oneLeech;
+
+    if(spellInfo.multiplier) {
+      return; // multiplier heals don't count for weights and they're adjusted out for total healing
+    }
+    this.totalAdjustedHealing += healVal.effective;
+    if(spellInfo.ignored) {
+      return; // ignored heals don't count for weights but they are part of adjusted total healing
+    }
+
+    if(healVal.overheal) {
+      return; // if a spell overheals, we know it couldn't have healed for more
+    }
+
+    const hotCount = target.activeBuffs()
+        .map(buffObj => buffObj.ability.guid)
+        .filter(buffId => getSpellInfo(buffId).masteryStack)
+        .length;
+
+    // INT //
+    let oneInt = 0;
+    if(spellInfo.int) {
+      const currInt = this._getCurrInt(); // TODO replace when dynamic stats
+      const bonusFromOneInt = (1 / currInt) * ARMOR_INT_MULTIPLIER;
+      oneInt = amount * bonusFromOneInt;
+    }
+    this.totalOneInt += oneInt;
+
+    // CRIT //
+    let oneCrit = 0;
+    if(spellInfo.crit) {
+      const currCritPerc = this.combatants.selected.critPercentage; // TODO replace when dynamic stats
+      const bonusFromOneCrit = 1 / 40000; // TODO replace when stat constants exist
+      if(spellInfo.isLivingSeed) {
```

Comment:
Adding a special bool for this feels like adding unnecessary complexity to me. How about just checking the spell id if it's living seed? You have to do special handling anyway, so moving all this special handling to here seems fine to me.

Reply:
My thought was to avoid directly using the event and offload everything to the spellInfo, but you're right that in this case that may be stupid.

Reply:
Changed as recommended.